pkg/image/apis/image/helper.go

```diff
@@ -523,6 +539,44 @@ func ImageWithMetadata(image *Image) error {
 	return nil
 }
 
+// FetchImageSignature attemps to fetch the signature blob for the provided
+// image and mutates the image object with the signature content if the registry
+// has defined a known signature store.
+func FetchImageSignature(registry string, image *Image) error {
+	for name, sigStore := range knownSignatureStores {
+		if registry != name {
+			continue
+		}
+		// TODO: Should we increase the trust to signature stores by verifying the HTTPS certificates?
+		client := &http.Client{
+			Transport: &http.Transport{
+				TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
```

Comment:
Yes, this should use ordinary, secure, HTTPS.  (To be consistent, with the same CA, private key, and `InsecureSkipVerify` opt-out flag as when accessing the docker/distribution registry).

Using secure HTTPS here is essential for the server to be able to quasi-revoke the signature by stopping to serve it; without that, an attacker could continue to serve a signature which the server wants to make impossible to use.

(i.e. the anti-replay facilities of HTTPS at this point are used instead of the TUF timestamping key.)